### PR TITLE
Removed classes to support PowerShell 4.0

### PR DESCRIPTION
**Pull request by bobbytreed**

### CODE REVIEW

File: GPRegistryPolicyParser.psm1 [hunk 1]
```diff
@@ -26,123 +27,76 @@ Import-LocalizedData  LocalizedData -filename GPRegistryPolicyParser.Strings.psd
...
-        return $Result
-    }
+    return $Result
 }
 
 Function New-GPRegistryPolicy
 {
     param (
-        [Parameter(Mandatory=$true,Position=0)]
+        [Parameter(Mandatory)]
```
Comment (Indhukrishna) [inline GPRegistryPolicyParser.psm1]: Why are the positional parameters removed?

File: GPRegistryPolicyParser.psm1 [hunk 2]
```diff
@@ -171,11 +125,11 @@ Specifies the path to the .pol file.
 .EXAMPLE
 C:\PS> Parse-PolFile -Path "C:\Registry.pol"
```
Comment (Indhukrishna) [inline GPRegistryPolicyParser.psm1]: The comment needs to be updated to match the function name

File: GPRegistryPolicyParser.psm1 [hunk 3]
```diff
@@ -449,6 +390,8 @@ Function Create-RegistrySettingsEntry
         [GPRegistryPolicy]
         $RegistryPolicy
     )
+
+    Write-Host "Creating key $($RP.KeyName)"
```
Comment (Indhukrishna) [inline GPRegistryPolicyParser.psm1]: Please try not to use Write-Host inside scripts http://www.jsnover.com/blog/2013/12/07/write-host-considered-harmful/

**Review by Indhukrishna — CHANGES_REQUESTED** [review]
Please update the code review with the requested changes.

File: GPRegistryPolicyParser.psm1 [hunk 4]
```diff
@@ -303,57 +260,56 @@ Reads registry policies from a list of entries.
...
-C:\PS> Read-RegistryPolicies -Division "LocalMachine" -Entries @('Software\Policies\Microsoft\Windows', 'Software\Policies\Microsoft\WindowsFirewall')
+C:\PS> Read-RegistryPolicies -Division "HKLM" -Entries @('Software\Policies\Microsoft\Windows', 'Software\Policies\Microsoft\WindowsFirewall')
 #>
 Function Read-RegistryPolicies
 {
     [OutputType([Array])]
     param (
 
-        [ValidateSet("LocalMachine", "CurrentUser", "Users")]
+        [ValidateSet("HKLM", "HKCU")]
```
Comment (zjalali) [inline GPRegistryPolicyParser.psm1:277]: >"HKLM", "HKCU" [](start = 21, length = 14)

This change will break GPRegistryPolicy module.

Comment (zjalali) [reply GPRegistryPolicyParser.psm1:277]: All changes in function signatures could break GPRegistryPolicy module.

---
In reply to: [121241843](https://github.com/PowerShell/GPRegistryPolicyParser/pull/1#discussion_r121241843) [](ancestors = 121241843)

File: GPRegistryPolicyParser.psm1 [hunk 5]
```diff
@@ -624,26 +568,20 @@ Function Get-RegKeyInfo
...
-        "String"       {
+        { @("String", "ExpandString") -contains $_ } {
             $Type = $RegKey.GetValueKind($ValueName)
             $Data = $RegKey.GetValue($ValueName)
             $Size = $Data.Length
         }
 
-        "ExpandString"       {
-            $Type = $RegKey.GetValueKind($ValueName)
-            $Data = $RegKey.GetValue($ValueName,$null,[Microsoft.Win32.RegistryValueOptions]::DoNotExpandEnvironmentNames)
```
Comment (zjalali) [inline GPRegistryPolicyParser.psm1:643]: >,[Microsoft.Win32.RegistryValueOptions]::DoNotExpandEnvironmentNames [](start = 53, length = 68)

This flag is required here. Why was it removed?

File: GPRegistryPolicyParser.psm1 [hunk 6]
```diff
@@ -692,22 +630,13 @@ Function IsRegistryKey
...
         [Microsoft.Win32.RegistryKey]
-        $Hive = [Microsoft.Win32.Registry]::LocalMachine
+        $Hive = [Microsoft.Win32.RegistryKey]::LocalMachine
     )
 
     $key = $Hive.OpenSubKey($Path)
 
     if ($key)
     {
-        if ($PSVersionTable.PSEdition -ieq 'Core')
```
Comment (zjalali) [inline GPRegistryPolicyParser.psm1:702]: >$PSVersionTable.PSEdition -ieq 'Core') [](start = 12, length = 38)

This coded was added for PS Core on NanoServer. Why was it removed?

**Comment by zjalali** [comment]
The `readme.md` file needs an update as well.

**Review by zjalali — CHANGES_REQUESTED** [review]
Please review the comments on the documents.

**Pull request closed without merging**